Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -55,10 +67,92 @@ extension OfflineMapAreasView {
             }
         }
         
-        /// Requests authorization to show notifications.
-        nonisolated func requestUserNotificationAuthorization() async {
-            _ = try? await UNUserNotificationCenter.current()
-                .requestAuthorization(options: [.alert, .sound])
+        /// Makes offline preplanned map models with information from the downloaded mobile map
+        /// packages for the online map.
+        func makeOfflinePreplannedMapModels() async {
+            guard let portalItemID else { return }
+            
+            let preplannedDirectory = FileManager.default.preplannedDirectory(forPortalItemID: portalItemID)
+            
+            guard let mapAreaIDs = try? FileManager.default.contentsOfDirectory(atPath: preplannedDirectory.path()) else { return }
+            
+            var mapAreas: [OfflinePreplannedMapArea] = []
+            
+            for mapAreaID in mapAreaIDs {
+                if let preplannedMapAreaID = PortalItem.ID(mapAreaID),
+                   let mapArea = await createMapArea(
+                    for: portalItemID,
+                    preplannedMapAreaID: preplannedMapAreaID
+                   ) {
+                    mapAreas.append(mapArea)
+                }
+            }
+            
+            offlinePreplannedMapModels = mapAreas.map { mapArea in
+                PreplannedMapModel(
+                    offlineMapTask: offlineMapTask,
+                    mapArea: mapArea,
+                    portalItemID: portalItemID,
+                    preplannedMapAreaID: mapArea.id!
+                )
+            }
+            .sorted(using: KeyPathComparator(\.preplannedMapArea.title))
+        }
+        
+        /// Creates a preplanned map area using a given portal item and map area ID to search for a corresponding
+        /// downloaded mobile map package. If the mobile map package is not found then `nil` is returned.
+        /// - Parameters:
+        ///   - portalItemID: The portal item ID.
+        ///   - preplannedMapAreaID: The preplanned map area ID.
+        /// - Returns: The preplanned map area.
+        private func createMapArea(
+            for portalItemID: PortalItem.ID,
+            preplannedMapAreaID: PortalItem.ID
+        ) async -> OfflinePreplannedMapArea? {
+            let fileURL = FileManager.default.preplannedDirectory(
+                forPortalItemID: portalItemID,
+                preplannedMapAreaID: preplannedMapAreaID
+            )
+            guard FileManager.default.fileExists(atPath: fileURL.path()) else { return nil }
+            // Make sure the directory is not empty because the directory will exist as soon as the
+            // job starts, so if the job fails, it will look like the mmpk was downloaded.
+            guard !FileManager.default.isDirectoryEmpty(atPath: fileURL) else { return nil }
```

Comment:
This check will be removed when the enhancement in `swift/5937` is implemented to download the mmpk to a temporary directory first which ensures that if the mmpk directory exists it will be non-empty.